pkg/versioned/versioned.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package versioned
+
+import (
+	"strconv"
+)
+
+// Version is the version used for each Object.
+type Version int64
+
+// ParseVersion parses the given string to Version, if the string is not
+// parsable a 0 Version is returned.
+// If the version is bigger than int64, it will return 9223372036854775807.
+func ParseVersion(s string) Version {
+	i, _ := strconv.ParseInt(s, 10, 64)
```

Comment:
What happens if someone sends us a garbage version that's not an integer? Where is this checked?

Reply:
```
// ParseVersion parses the given string to Version, if the string is not
// parsable a 0 Version is returned.
```

Reply:
Somehow I missed that, woops! :-)